proposals/000-ensure-compilation-scalability.md

```diff
@@ -1,56 +1,99 @@
 
-1. A tightening of the compiler's requirements such that only symbols explicitly referenced in `import` statements in a `.scala` file are required on the classpath during compilation.
-2. A dependency story that is sufficiently simplified to allow for a scalac/dotc integrated tool that extracts all direct symbol dependencies of a Scala source file *without* a classpath. Such a tool would be the holy grail for Scala compilation scalability, as it would (given a lookup table from symbol to classpath entry) allow for very-fine-grained, semi-automatic dependency inference and build graph partitioning.
-3. An expansion of the Scala Language Specification to list all cases in which a symbol from another compilation unit must be present on the classpath, including: 1) subclassing, 2) return types of superclasses' public methods, 3) direct reference, 4) etc.
-4. Changes to scalac/dotc internal APIs to make expansion of transitive symbols explicit, in order to bias compiler developers against unnecessary transitive walks.
-5. A compiler flag to require `import` statements for all symbols used during compilation (including those not otherwise mentioned in the source). This would primarily assist with making the transition from transitive to direct dependencies, rather than helping to maintain a build that is already using direct dependencies. (via [@posco](twitter.com/posco))
+```scala
+// A.scala
+class A
 
-## Cost
+// B.scala
+class B extends A
+
+// C.scala
+object C extends App {
+  println(new B)
+}
+```
+
+In this updated example, `B` no longer uses `A` in its constructor, but instead subclasses it. Even though, `C` still doesn't use `A` directly, we now have to provide `A` on the compilation classpath, otherwise an error occurs.
+
+```
+$ scalac -d a A.scala
+
+$ scalac -d b -cp a B.scala
+
+$ scalac -d c -cp b C.scala
+error: missing or invalid dependency detected while loading class file 'B.class'.
+Could not access type A in package <empty>,
+because it (or its dependencies) are missing. Check your build definition for
+missing or conflicting dependencies. (Re-run with `-Ylog-classpath` to see the problematic classpath.)
+A full rebuild may help if 'B.class' was compiled against an incompatible version of <empty>.
+one error found
+```
 
-Although this proposal is bounded in scope (it is possible to call it "done" when one or more of the above proposals are implemented), it is purposely open-ended with regard to strategy in order to give the implementers flexibility to work with the core team, and to use their best judgement.
+We can observe that the Scala compiler: 1) exhibits unintuitive behavior when faced with seemingly minor code changes,
+2) produces confusing error messages that don't provide instructions how to fix underlying errors. This proposal aims at improving user experience in these areas.
+
+## Statement
```

Comment:
I feel that this hidden beneath/behind a lot of explanation... would like to either reduce the amount of background, or try to highlight the statement more strongly, as I feel that it stands alone with relatively little explanation.